cucumber/authenticators_jwt/features/authn_jwt_fetch_signing_key.feature

```diff
@@ -658,3 +658,166 @@ Feature: JWT Authenticator - Fetch signing key
+    Then host "myapp" has been authorized by Conjur
+    And The following appears in the log after my savepoint:
+    """
+    cucumber:host:myapp successfully authenticated with authenticator authn-jwt service cucumber:webservice:conjur/authn-jwt/raw
+    """
+
+  Scenario: ONYX-15323: public-keys with invalid issuer variable
+    Given I load a policy:
+    """
+    - !policy
+      id: conjur/authn-jwt/raw
+      body:
+      - !webservice
+      - !variable public-keys
+      - !variable issuer
+      - !variable token-app-property
+
+      - !group hosts
+
+      - !permit
+        role: !group hosts
+        privilege: [ read, authenticate ]
+        resource: !webservice
+
+    - !host
+      id: myapp
+      annotations:
+        authn-jwt/raw/project_id: myproject
+
+    - !grant
+      role: !group conjur/authn-jwt/raw/hosts
+      member: !host myapp
+    """
+    And I am the super-user
+    And I initialize remote JWKS endpoint with file "public-key-2" and alg "RS256"
+    And I successfully set authn-jwt public-keys variable to value from remote JWKS endpoint "public-key-2" and alg "RS256"
+    And I successfully set authn-jwt "issuer" variable to value "invalid-issuer"
+    And I successfully set authn-jwt "token-app-property" variable to value "host"
+    And I am using file "public-key-2" and alg "RS256" for remotely issue token:
+    """
+    {
+      "host":"myapp",
+      "project_id": "myproject",
+      "iss": "valid-issuer"
+    }
+    """
+    And I save my place in the log file
+    When I authenticate via authn-jwt with the JWT token
+    Then the HTTP response status code is 401
+    And The following appears in the log after my savepoint:
+    """
+    CONJ00035E Failed to decode token (3rdPartyError ='#<JWT::InvalidIssuerError: Invalid issuer. Expected invalid-issuer, received valid-issuer>')>
+    """
+
+  Scenario: ONYX-15324: public-keys with valid issuer, token is signed by other key
+    Given I load a policy:
+    """
+    - !policy
+      id: conjur/authn-jwt/raw
+      body:
+      - !webservice
+      - !variable public-keys
+      - !variable issuer
+      - !variable token-app-property
+
+      - !group hosts
+
+      - !permit
+        role: !group hosts
+        privilege: [ read, authenticate ]
+        resource: !webservice
+
+    - !host
+      id: myapp
+      annotations:
+        authn-jwt/raw/project_id: myproject
+
+    - !grant
+      role: !group conjur/authn-jwt/raw/hosts
+      member: !host myapp
+    """
+    And I am the super-user
+    And I initialize remote JWKS endpoint with file "public-key-1" and alg "RS256"
+    And I successfully set authn-jwt public-keys variable to value from remote JWKS endpoint "public-key-1" and alg "RS256"
+    And I successfully set authn-jwt "issuer" variable to value "valid-issuer"
+    And I successfully set authn-jwt "token-app-property" variable to value "host"
+    And I am using file "public-key-2" and alg "RS256" for remotely issue token:
```

Comment:
If you will run in the future this test only you will not have public-key-2 because you create it on ONYX-15323 test

Reply:
I'll add init sentence above for better visibility